warehouse/manage/views.py

```diff
@@ -3073,6 +3073,9 @@ def manage_project_oidc_publishers(self):
         if not self.oidc_enabled:
             raise HTTPNotFound
 
+        if not self.request.user.in_oidc_beta and not self.project.oidc_providers:
```

Comment:
We need to make sure that in the case where the user isn't in the beta, but there are publishers, we aren't showing the "Add a new publisher" form (or that it's displayed as disabled at least). Same for the "Remove" button for existing publisher(s).

Reply:
Good catch! Yeah, those forms will currently 403 on submission for those users, but I'll explicitly disable them with CSS as well. That way only a motivated user would hit the 403.